### ISSUE TITLE
Port PathSeq from HoTT-Agda to TypeTopology

### ISSUE THREAD

**Issue by ettorealdrovandi** [martinescardo/TypeTopology#107]
Request to port the `PathSeq` functionality from the
`PathSeq` library in https://github.com/HoTT/HoTT-Agda to
`TypeTopology`. In HoTT-Agda the data structure `PathSeq` is defined in `lib.Base`,  with auxiliary structure and data in `lib.path-seq.*`.

Let us call a term `p : x ＝ y` a "path" from `x` to `y`. Abstract
path manipulation should have the following features:

1. A data structure to represent a path normalized concatenation. This is called an "abstract path."  
   
   If `p`, `q`, `r`, `s`, `t` are paths, then, say, `[p,q,r,s,t]` should be the corresponding abstract path, regardless of the parentheses. That is, `p ∙ (q ∙ (r ∙ (s ∙ t)))` and `(p ∙ q) ∙ (r ∙ (s ∙ t)) ` should ideally give rise to the same `[p,q,r,s,t]`.

2. Path reasoning: equational-style reasoning for abstract path concatenation.
3. Conversion to and insertion from paths.
4. Abstract path equality, say, `＝＝`
5. Manipulation: if two abstract paths `s` (resp. `t`) contain
   sub-(abstract) paths `u` (resp. `v`) and `α : u ＝＝ v`, we should
   be able to write somethings of the form:

```agda
   lemma : s ＝＝ t
   lemma = s ＝＝⟨⟨ insertion-point | end-point | α ⟩⟩
           t ＝＝∎∎ 
```
    
In the above the insertion and the end points specify where `u` occur
in `s`. This should work even if `u` and `v` are of not of the same
length.

**Comment by ayberkt** [comment]
Are there any plans to implement this?

**Comment by ayberkt** [comment]
@ettorealdrovandi @martinescardo Is this what has been implemented in PR #205? Sorry I didn’t follow all the discussion. If that is the case, can we close this issue?
